File: app/src/main/java/io/agora/metagpt/adapter/ChatMessageAdapter.java
package io.agora.metagpt.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.graphics.Rect;
import android.text.Html;
import android.text.Spanned;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import io.agora.metagpt.R;
import io.agora.metagpt.models.ChatMessageModel;


public class ChatMessageAdapter extends RecyclerView.Adapter<ChatMessageAdapter.MyViewHolder> implements IMessageView{
    private final Context mContext;
    private final List<ChatMessageModel> mDataList;

    public ChatMessageAdapter(Context context, List<ChatMessageModel> list) {
        this.mContext = context;
        this.mDataList = list;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(mContext).inflate(R.layout.item_chat_message, parent, false);
        return new MyViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, @SuppressLint("RecyclerView") int position) {
        if (null != mDataList) {
            ChatMessageModel chatMessage = mDataList.get(position);
            String text= mContext.getResources().getString(R.string.message_font_text, chatMessage.getUsername(),chatMessage.getMessage());
            Spanned colorText = Html.fromHtml(text);
            holder.tvMessage.setText(colorText);
        }
    }

    @Override
    public int getItemCount() {
        if (null == mDataList) {
            return 0;
        }
        return mDataList.size();
    }

    public List<ChatMessageModel> getDataList() {
        return mDataList;
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        private final TextView tvMessage;

        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            tvMessage = itemView.findViewById(R.id.tvMessage);
        }
    }


    public static class SpacesItemDecoration extends RecyclerView.ItemDecoration {
        private final int space;

        public SpacesItemDecoration(int space) {
            this.space = space;
        }

        @Override
        public void getItemOffsets(Rect outRect, @NonNull View view,
                                   RecyclerView parent, @NonNull RecyclerView.State state) {
            outRect.left = 0;
            outRect.right = 0;
            outRect.bottom = space;

            // Add top margin only for the first item to avoid double space between items
            if (parent.getChildAdapterPosition(view) == 0)
                outRect.top = space;
        }
    }
}

File: app/src/main/java/io/agora/gpt/ui/adapter/ChatMessageAdapter.kt
package io.agora.gpt.ui.adapter

import android.annotation.SuppressLint
import android.content.Context
import android.graphics.Color
import android.graphics.Typeface
import android.text.Spannable
import android.text.SpannableStringBuilder
import android.text.style.ForegroundColorSpan
import android.text.style.StyleSpan
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import io.agora.gpt.R
import io.agora.gpt.ui.main.ChatMessageModel

class ChatMessageAdapter constructor(private val mContext: Context, val dataList: List<ChatMessageModel>) :
    RecyclerView.Adapter<MyViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val view = LayoutInflater.from(mContext).inflate(R.layout.item_chat_message, parent, false)
        return MyViewHolder(view)
    }

    override fun getItemCount(): Int {
        return dataList.size
    }

    override fun onBindViewHolder(holder: MyViewHolder, @SuppressLint("RecyclerView") position: Int) {
        val chatMessageModel = dataList[position]
        val ssb = SpannableStringBuilder()
        val name = "${chatMessageModel.name} : "
        ssb.append(name)
        ssb.append(chatMessageModel.message)
        //设置字体颜色
        ssb.setSpan(
            ForegroundColorSpan(Color.parseColor("#A6C4FF")),
            0,
            name.length,
            Spannable.SPAN_EXCLUSIVE_EXCLUSIVE
        )
        ssb.setSpan(StyleSpan(Typeface.BOLD), 0, name.length, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE)
        holder.tvMessage.text = ssb
    }
}


class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
    val tvMessage: TextView

    init {
        tvMessage = itemView.findViewById(R.id.tvMessage)
    }
}